bls/src/main/java/tech/pegasys/teku/bls/BLS.java

```diff
@@ -94,15 +105,29 @@ public static BLSSignature aggregate(final Stream<BLSSignature> signatures) {
    */
   public static boolean aggregateVerify(
       List<BLSPublicKey> publicKeys, List<Bytes> messages, BLSSignature signature) {
+    checkArgument(
+        publicKeys.size() == messages.size(),
+        "Number of public keys and number of messages differs.");
+    if (publicKeys.isEmpty()) {
+      return false;
+    }
+    // Check that there are no duplicate messages
+    Set<Bytes> set = new HashSet<>();
+    for (Bytes message : messages) {
+      if (!set.add(message)) return false;
+    }
```

Comment:
```suggestion
    if (new HashSet(messages).size() != messages.size) return false;
```
Minor code size improvement

Reply:
This is similar to how I first wrote it, but my ever-optimising brain thought that it might be better to stop building the HashSet early if we found a duplicate. But our sets will be small, and your suggestion is elegant, so I will adopt it 😃